services/awp_azure_onboarding/awp_azure_onboarding.go

```diff
@@ -0,0 +1,209 @@
+	MissingAwpPrivateNetworkRegions *[]string                 `json:"missingAwpPrivateNetworkRegions"`
+	AccountIssues                   *AccountIssues            `json:"accountIssues"`
+	CloudAccountId                  string                    `json:"cloudAccountId"`
+	AgentlessProtectionEnabled      bool                      `json:"agentlessProtectionEnabled"`
+	ScanMode                        string                    `json:"scanMode"`
+	Provider                        string                    `json:"provider"`
+	ShouldUpdate                    bool                      `json:"shouldUpdate"`
+	IsOrgOnboarding                 bool                      `json:"isOrgOnboarding"`
+	CentralizedCloudAccountId       string                    `json:"centralizedCloudAccountId"`
+}
+
+type CreateOptions struct {
+	ShouldCreatePolicy string `url:"shouldCreatePolicy"`
+}
+
+func (service *Service) CreateAWPOnboarding(id string, req CreateAWPOnboardingRequest, queryParams CreateOptions) (*http.Response, error) {
+	// Define the maximum number of retries and the interval between retries
+	maxRetries := 3
+	retryInterval := time.Second * 5
+
+	// Determine the base path based on ScanMode
+    var basePath string
+    switch req.ScanMode {
+    case ScanModeInAccountSub:
+        basePath = fmt.Sprintf("%s/%s/enableSubAccount", azureOnboardingResourcePath, id)
+    case ScanModeInAccountHub:
+        basePath = fmt.Sprintf("%s/%s/enableCentralizedAccount", azureOnboardingResourcePath, id)
+    default:
+        basePath = fmt.Sprintf("%s/%s/enable", azureOnboardingResourcePath, id)
+    }
+
+	// Initialize the response and error variables outside the loop
+	var resp *http.Response
+	var err error
+
+	// Attempt the request up to maxRetries times
+	for i := 0; i < maxRetries; i++ {
+		// Make the request
+		resp, err = service.Client.NewRequestDo("POST", basePath, queryParams, req, nil)
+		if err == nil {
+			// If the request was successful, return the response
+			return resp, nil
+		}
+
+		// If the request failed with a 404 status code, wait for the retry interval before trying again
+		if resp != nil && resp.StatusCode == 404 {
+			time.Sleep(retryInterval)
+		} else {
+			// If the status code is not 404, return the response and error immediately
+			return resp, err
+		}
+	}
+
+	// If the function hasn't returned after maxRetries, return an error
+	return nil, fmt.Errorf("failed to create AWP Onboarding after %d attempts: %w", maxRetries, err)
+}
+
+func (service *Service) GetAWPOnboarding(cloudProvider, id string) (*GetAWPOnboardingResponse, *http.Response, error) {
+	v := new(GetAWPOnboardingResponse)
+	path := fmt.Sprintf("workload/agentless/%s/accounts/%s", cloudProvider, id)
+	resp, err := service.Client.NewRequestDo("GET", path, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+	return v, resp, nil
+}
+
+func (service *Service) DeleteAWPOnboarding(id string) (*http.Response, error) {
```

Comment:
basically also part of the logic can be common,

@bzemms what do you think?

here and maybe in more CRUD functions were logic is the same and only the parameter type and url are different, it can be unified by accepting something like: req Interface{}